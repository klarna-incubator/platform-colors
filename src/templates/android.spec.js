const generate = require('./android');
const parseColorManifest = require('../parse-color-manifest');
const lightColors = parseColorManifest(require('./fixtures/light-colors.json'));
const darkColors = parseColorManifest(require('./fixtures/dark-colors.json'));

describe('android template', () => {
  it.each([
    ['light', lightColors],
    ['dark', darkColors],
  ])('generates %s colors', (name, colors) => {
    expect(generate(colors)).toMatchSnapshot();
  });
});
