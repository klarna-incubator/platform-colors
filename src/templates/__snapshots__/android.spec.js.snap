// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android template generates dark colors 1`] = `
Array [
  Array [
    "values/colors.xml",
    "<?xml version=\\"1.0\\"?>
<resources>
  <!--Generated code starts changes will be overridden-->
  <color name=\\"rnpc_string_color\\">#ffffff</color>
  <color name=\\"rnpc_alpha_color\\">#80ff0000</color>
  <!--Generated code ends-->
  <color name=\\"background\\">#ffffff</color>
  <color name=\\"text\\">#000000</color>
  <color name=\\"accent\\">#ffc0cb</color>
  <color name=\\"contrasted\\">#99cccccc</color>
</resources>",
  ],
  Array [
    "values-night/colors.xml",
    "<?xml version=\\"1.0\\"?>
<resources>
  <!--Generated code starts changes will be overridden-->
  <color name=\\"rnpc_string_color\\">#000000</color>
  <color name=\\"rnpc_alpha_color\\">#80ffc8c8</color>
  <!--Generated code ends-->
  <color name=\\"background\\">#000000</color>
  <color name=\\"text\\">#ffffff</color>
  <color name=\\"contrasted\\">#99333333</color>
</resources>",
  ],
]
`;

exports[`android template generates light colors 1`] = `
Array [
  Array [
    "values/colors.xml",
    "<?xml version=\\"1.0\\"?>
<resources>
  <!--Generated code starts changes will be overridden-->
  <color name=\\"rnpc_string_color\\">#ffffff</color>
  <color name=\\"rnpc_alpha_color\\">#80ff0000</color>
  <color name=\\"rnpc_named_color\\">#0000ff</color>
  <!--Generated code ends-->
  <color name=\\"background\\">#ffffff</color>
  <color name=\\"text\\">#000000</color>
  <color name=\\"accent\\">#ffc0cb</color>
  <color name=\\"contrasted\\">#99cccccc</color>
</resources>",
  ],
]
`;
